runtime/src/fees.rs

```diff
@@ -71,21 +71,25 @@ decl_module! {
 impl<T: Trait> Module<T> {
     /// Called by any other module who wants to trigger a fee payment for a given account.
     /// The current fee price can be retrieved via Fees::price_of()
-    pub fn pay_fee(who: T::AccountId, key: T::Hash) -> DispatchResult {
+    pub fn pay_fee(from: T::AccountId, to: T::AccountId, key: T::Hash) -> DispatchResult {
         ensure!(<Fees<T>>::contains_key(key), "fee not found for key");
 
         let single_fee = <Fees<T>>::get(key);
-        Self::pay_fee_given(who, single_fee.price)?;
+        Self::pay_fee_given_to(from, to, single_fee.price)?;
 
         Ok(())
     }
 
     /// Pay the given fee
-    pub fn pay_fee_given(who: T::AccountId, fee: T::Balance) -> DispatchResult {
-        let _ = <pallet_balances::Module<T> as Currency<_>>::withdraw(
-            &who,
+    pub fn pay_fee_given_to(
+        from: T::AccountId,
+        to: T::AccountId,
+        fee: T::Balance,
+    ) -> DispatchResult {
+        let _ = <pallet_balances::Module<T> as Currency<_>>::transfer(
```

Comment:
Is there a better way than using a transfer for fees? The transfer looks in Polkascan like the anchorer did intentionally transfer to another party, while in reality the intent was to just pay fees. 

Maybe withdraw and deposit instead?

Reply:
Wouldn't withdraw and deposit appear in the polkscan?